### PR TITLE
Adding ROM version

### PR DESCRIPTION
**Pull request by nquarton**
- Change FW_REV_ID[0] 15:0 to be the ROM version
- Change FMC version to be 16 bits and shift to FW_REV_ID[0] 31:16

### CODE REVIEW

File: builder/src/version.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Licensed under the Apache-2.0 license
+
+pub const ROM_VERSION_MAJOR: u16 = 1;
```
Comment (korran) [inline builder/src/version.rs:3]: Do we want this in the source code, or fill this as part of the release process?

Comment (nquarton) [reply builder/src/version.rs:3]: This is something I am looking for input on. If you have an idea for a way we can populate this during a release I think that would be better. You could in theory try to pull it from a tag but I was thinking that may be fragile and requires the tag to be populated before the release is made.

File: builder/src/version.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+pub const ROM_VERSION_MINOR: u16 = 0;
+pub const ROM_VERSION_PATCH: u16 = 1;
+
+// ROM Version - 16 bits
```
Comment (nquarton) [inline builder/src/version.rs:7]: We mentioned the version source could be changed down the road, but just wanted to call out we won't want to change the bit allocation for these 16 bits once we merge. So we should at least make sure we are all in agreement on that.

**Review by JohnTraverAmd — APPROVED** [review]

File: builder/src/version.rs (hunk 1 above)
Comment (FerralCoder) [reply builder/src/version.rs:3]: In the interest of getting this patch out quickly, I suggest we leave it like this right now, and look for a better method for the next release

**Review by FerralCoder — APPROVED** [review]

File: rom/dev/src/main.rs [hunk 3]
```diff
@@ -110,6 +110,10 @@ pub extern "C" fn rom_entry() -> ! {
         }
     );
 
+    // Set the ROM version
+    let rom_info = unsafe { &CALIPTRA_ROM_INFO };
+    env.soc_ifc.set_rom_fw_rev_id(rom_info.version);
```
Comment (mhatrevi) [inline rom/dev/src/main.rs:115]: Can we add a test in rom/tests (choose the appropriate module in that) to test that the version is set correctly?

**Review by mhatrevi — CHANGES_REQUESTED** [review]

File: rom/dev/src/main.rs (hunk 3 above)
Comment (nquarton) [reply rom/dev/src/main.rs:115]: This is a good point. This test is missing in ROM. I'll add it.

Comment (nquarton) [reply rom/dev/src/main.rs:115]: This has been added now

File: builder/src/version.rs (hunk 2 above)
Comment (nquarton) [reply builder/src/version.rs:7]: It sounds like everyone is agreed this is a good format

**Review by mhatrevi — APPROVED** [review]

**Review by jhand2 — APPROVED** [review]

**Pull request merged**